admin-ui/src/component/my-units/MyUnitRecurringReservation/MyUnitRecurringReservation.tsx

```diff
@@ -0,0 +1,99 @@
+import { H1 } from "common/src/common/typography";
+import { Button, IconAngleLeft } from "hds-react";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+import styled from "styled-components";
+import { myUnitUrl } from "../../../common/urls";
+import { Container } from "../../../styles/layout";
+import { BasicLink } from "../../../styles/util";
+import Loader from "../../Loader";
+import withMainMenu from "../../withMainMenu";
+import { MyUnitRecurringReservationForm } from "./MyUnitRecurringReservationForm";
+import type { ReservationsMade } from "./RecurringReservationDone";
+import RecurringSuccess from "./RecurringReservationDone";
+import { useRecurringReservationsUnits } from "./hooks";
+
+const PreviousLinkWrapper = styled.div`
+  padding: var(--spacing-xs);
+`;
+
+type Params = {
+  unitId: string;
+  reservationUnitId: string;
+};
+
+const BackLinkHeader = ({ unitId }: { unitId: number }) => {
+  const { t } = useTranslation();
+  const previousUrl = myUnitUrl(unitId);
+
+  return (
+    <PreviousLinkWrapper>
+      <BasicLink to={previousUrl}>
+        <Button
+          aria-label={t("common.prev")}
+          size="small"
+          variant="supplementary"
+          iconLeft={<IconAngleLeft />}
+        >
+          {t("common.prev")}
+        </Button>
+      </BasicLink>
+    </PreviousLinkWrapper>
+  );
+};
+
+const MyUnitRecurringReservation = ({ unitId }: { unitId: number }) => {
+  const [reservationMade, setReservationMade] =
+    useState<ReservationsMade | null>(null);
+  const { t } = useTranslation();
+
+  const { loading, reservationUnits } = useRecurringReservationsUnits(unitId);
+
+  if (loading) return <Loader />;
+
+  return (
+    <>
+      <BackLinkHeader unitId={unitId} />
+      <Container>
+        {reservationMade !== null ? (
+          <RecurringSuccess reservation={reservationMade} />
+        ) : (
+          <>
+            <H1 $legacy>{t("MyUnits.RecurringReservation.pageTitle")}</H1>
+            {reservationUnits != null && reservationUnits.length > 0 ? (
```

Comment:
Using just `reservationUnits?.length > 0` might make it more readable.

Reply:
valid JS but TS has a hissy fit about it (both VS code with TS 5.1 and TS playground with 4.9.5). undefined > 0 is not valid.